os/dutils.sh

```diff
@@ -57,6 +60,63 @@ function FIND_DBGBINFILE()
 	KERNEL_DBG_BIN="tinyara$EXEEXT"
 }
 
+function TRAP_MENU()
+{
+	echo =============================================================================
+	echo "  \"Select binary to debug crash with TRAP tool\""
+	echo =============================================================================
+	echo "  \"1. Default binary from previous build at build/output/bin location\""
+	echo "  \"2. Custom binary at any location\""
+	echo "  \"x. Exit\""
+	echo =============================================================================
+
+	read SELECT_BIN
+	case ${SELECT_BIN,,} in
+		1|default)
+			if [ -z "$(ls -A $BINDIR)" ]; then
+				echo "No output file, Build the code and run TRAP again."
+				exit 1
+			fi
+			echo "Enter the crash log file name: (ex: ../tools/trap/testlogs)"
+			read LOG_FILE
+			while [ ! -f ${LOG_FILE} ]; do
+				echo "$LOG_FILE: No such file, try again"
+				read LOG_FILE
+			done
+			TRAP_RUN log $LOG_FILE
+			;;
+		2|manual)
+			echo "Enter the following wrt os directory"
+			echo ""
+			echo "Enter the binary folder name: (ex: ../../build/output/bin/)"
+			read EXT_BIN_DIR
+			while [ -z "$(ls -A $EXT_BIN_DIR)" ]; do
```

Comment:
If we are using a while loop, the user is forced to enter the proper file. He does not have a method to exit if he does not have the required file.

Reply:
Updated to go to previous menu